core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -490,6 +574,19 @@ public void updateSettings(Settings settings) {
         }
     }
 
+    private void validate(Map<String, Settings> groupSettings) {
+        for (String key : groupSettings.keySet()) {
+            if (!THREAD_POOL_TYPES.containsKey(key)) {
+                continue;
+            }
+            String type = groupSettings.get(key).get("type");
+            // TODO: the type equality check can be removed after #3760/#6732 are addressed
+            if (type != null && !THREAD_POOL_TYPES.get(key).getType().equals(type)) {
+                throw new IllegalArgumentException("setting " + THREADPOOL_GROUP + key + ".type to " + type + " is not permitted");
```

Comment:
can the error message mention the only allowed value, so that users with existing settings can update?